AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java

```diff
@@ -190,14 +199,39 @@ public void afterTextChanged(final Editable editable) {
                 setupEnter.setTextColor(ContextCompat.getColor(this, R.color.colorAccent));
             } else {
                 setupEnter.setTextColor(ContextCompat.getColor(this, R.color.textbox_secondary));
+                final InputMethodManager inputMethodManager =
+                        (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
+                inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
             }
         });
 
         joinButton = findViewById(R.id.setup_button);
         joinButtonText = findViewById(R.id.setup_button_text);
+
+        final ToggleButton deviceOptionsButton = findViewById(R.id.device);
+        deviceOptionsButton.setOnFocusChangeListener((view, hasFocus) -> {
+            if (hasFocus) {
+                openAudioDeviceList();
+            }
+        });
+        deviceOptionsButton.setOnClickListener(l -> {
+            openAudioDeviceList();
+        });
+
         hidePermissionsWarning();
     }
 
+    private void openAudioDeviceList() {
+        if (audioDeviceSelectionPopupWindow == null) {
+            final AudioSessionManager audioSessionManager
+                    = ((AzureCalling) getApplicationContext()).getAudioSessionManager();
+            audioDeviceSelectionPopupWindow = new AudioDeviceSelectionPopupWindow(this, audioSessionManager);
+        } else {
```

Comment:
If `showAtLocation` is not called from the constructor, the `else` can be removed.

Reply:
addressed.